src/domain/faucet/store/builder/store.builder.ts

```diff
@@ -0,0 +1,93 @@
+import type { Store, AnyAction } from 'redux'
+import { EventBus } from 'ts-bus'
+import { initFaucetEventListeners } from 'adapters/faucet/primary/eventListeners'
+import { UnspecifiedError } from 'domain/faucet/entity/error'
+import type { DeepReadonly } from 'superTypes'
+import type { AppState } from '../appState'
+import type { Dependencies } from '../store'
+import { configureStore } from '../store'
+
+export type FaucetStoreParameters = {
+  dependencies: Dependencies | null
+  eventBus: EventBus | null
+  preloadedState?: AppState
+}
+
+export class FaucetStoreBuilder {
+  private readonly faucetStoreParameters: FaucetStoreParameters
+
+  constructor(
+    faucetStoreParameters: DeepReadonly<FaucetStoreParameters> = {
+      dependencies: null,
+      eventBus: null
+    }
+  ) {
+    this.faucetStoreParameters = faucetStoreParameters
+  }
+
+  public withDependencies(dependencies: Dependencies): FaucetStoreBuilder {
+    if (!this.isDependencies(dependencies)) {
+      throw new UnspecifiedError(
+        'Ooops... Dependencies must be provided to build a Faucet store...'
+      )
+    }
+    return new FaucetStoreBuilder({ ...this.faucetStoreParameters, dependencies })
+  }
+
+  public withEventBus(eventBus: DeepReadonly<EventBus>): FaucetStoreBuilder {
+    if (!(eventBus instanceof EventBus)) {
```

Comment:
As we are using this instance check twice, maybe with could make a function of it? As `isDependencies `?

Reply:
Here we use a simple instanceof based assertion so I'm not sure transforming it to a type-guard is relevant.
isDependencies type guard is much more relevant as it's harder to test 'faucetGateway' in value predicate directly in the if statement.

Reply:
Yes you're right, but I think we could keep this assertion as existing, to mainly prevent from adding a method per assertion in each store builder, if you are OK with that